docs/architecture.md

```diff
@@ -94,12 +92,10 @@ def my_command(ctx, pulp_ctx):
     ctx.obj = MyEntityContext(pulp_ctx)
 ```
 
-To declare version restrictions on *options* (*body parameters* in the API), the [`preprocess_entity`][pulp_glue.common.context.PulpEntityContext.preprocess_entity]
-method can check if a given option is being used and conditionally apply the requirements to the context.
+To declare version restrictions on *options*, the [`preprocess_entity`][pulp_glue.common.context.PulpEntityContext.preprocess_entity] method can be used to check if a given option is present in the request body and conditionally apply the requirements to the context.
 
-Each *plugin* may already have `Contexts` binded to specific [resource_class](/using_the_cli/#general-command-syntax) entities,
-which can be found under `pulp_glue/{plugin-name}/context.py`. In the example, a guard is added to `my_option`, which belongs to the
-`Repository` *resource_class* of `"my_plugin"`.
+Each *plugin* have `Contexts` binded to specific [resource_class](using_the_cli.md/#general-command-syntax) entities, which can be found under `pulp_glue/{plugin-name}/context.py`.
```

Comment:
I would say this is a related but a complementary information:
-  line 46 is mostly useful for adding a new plugin, where it will be necessary to subclass `PulpEntityContext` and setup the context.
- this line is mostly useful for someone (like me) who wants to modify an existing Context and won't do a deep dive into plumbing. It maybe implicit from the architecture that existing plugins already have Context bound to resources, and it's possible to guess that lives it there, but IMHO its not so obvious for that audience.

Maybe it would be appropriate to move it there:
> Further we encourage the handling of communication with certain endpoints by subclassing the `PulpEntityContext`.
> Existing plugins already have *Contexts* bound to specific *resource classes*, which can be found under `pulp_glue/{plugin-name}/context.py`.

Reply:
It's always hard to make any definite claim about coverage in the cli. You see, we add what we need when we need it. This will probably never be complete.